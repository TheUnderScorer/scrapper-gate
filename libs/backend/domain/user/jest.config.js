const dotenv = require('dotenv');

dotenv.config({
  path: `${process.cwd()}/.local.env`,
});

module.exports = {
  displayName: 'backend-domain-user',
  preset: '../../../../jest.preset.js',
  globals: {
    'ts-jest': {
      tsConfig: '<rootDir>/tsconfig.spec.json',
    },
  },
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/backend/domain/user',
  setupFilesAfterEnv: [`${process.cwd()}/tests/setupBackend.ts`],
};
