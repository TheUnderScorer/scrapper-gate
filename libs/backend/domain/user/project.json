{
  "root": "libs/backend/domain/user",
  "sourceRoot": "libs/backend/domain/user/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/backend/domain/user/**/*.ts"]
      },
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/backend/domain/user"],
      "options": {
        "jestConfig": "libs/backend/domain/user/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
