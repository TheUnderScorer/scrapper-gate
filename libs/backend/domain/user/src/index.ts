export * from './models/User.model';
export * from './repositories/User.repository';
export * from './user.entity';
export * from './handlers/createUser.handler';
export * from './commands/CreateUser.command';
export * from './events/UserCreated.event';
