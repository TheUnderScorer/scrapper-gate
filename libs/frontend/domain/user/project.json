{
  "root": "libs/frontend/domain/user",
  "sourceRoot": "libs/frontend/domain/user/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/frontend/domain/user/**/*.{ts,tsx,js,jsx}"]
      },
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/frontend/domain/user"],
      "options": {
        "jestConfig": "libs/frontend/domain/user/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
