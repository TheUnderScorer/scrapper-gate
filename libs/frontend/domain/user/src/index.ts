export * from './components/UserAvatar/UserAvatar';
