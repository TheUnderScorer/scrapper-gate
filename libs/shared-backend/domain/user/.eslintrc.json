{
  "extends": ["../../../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": ["libs/shared-backend/domain/user/tsconfig.*?.json"]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
