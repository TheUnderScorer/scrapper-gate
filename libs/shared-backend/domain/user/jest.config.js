const dotenv = require('dotenv');

dotenv.config({
  path: `${process.cwd()}/.local.env`,
});

module.exports = {
  displayName: 'shared-backend-domain-user',
  preset: '../../../../jest.preset.js',
  globals: {
    'ts-jest': {
      tsConfig: '<rootDir>/tsconfig.spec.json',
    },
  },
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/shared-backend/domain/user',
  setupFilesAfterEnv: [`${process.cwd()}/tests/setupBackend.ts`],
};
