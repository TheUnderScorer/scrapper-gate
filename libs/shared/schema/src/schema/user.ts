import gql from 'graphql-tag';

export const userSchema = gql`
  type User implements BaseEntity {
    id: ID!
    createdAt: Date!
    updatedAt: Date!
    firstName: String
    lastName: String
    email: String!
  }

  extend type Query {
    me: User
  }
`;
