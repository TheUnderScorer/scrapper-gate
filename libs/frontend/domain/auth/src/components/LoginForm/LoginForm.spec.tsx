import {
  CreateUserDocument,
  LoginDocument,
} from '@scrapper-gate/frontend/schema';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import {
  CreateUserMutation,
  CreateUserMutationVariables,
  LoginMutation,
  LoginMutationVariables,
} from '@scrapper-gate/shared/schema';
import {
  LoginForm,
  LoginFormProps,
  LoginFormType,
  useTokensStore,
} from '@scrapper-gate/frontend/domain/auth';
import { act, render, RenderResult, waitFor } from '@testing-library/react';
import fireEvent from '@testing-library/user-event';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { createMockUser } from '@scrapper-gate/shared/domain/user';

const user = createMockUser();

const createUserResult = {
  tokens: {
    accessToken: '#token',
    refreshToken: '#refresh_token',
  },
  user,
};

const mocks: MockedResponse[] = [
  {
    request: {
      query: LoginDocument,
      variables: {
        input: {
          username: 'test@test.com',
          password: 'password',
        },
      } as LoginMutationVariables,
    },
    result: {
      data: {
        login: {
          accessToken: '#token',
          refreshToken: '#refresh_token',
        },
      } as LoginMutation,
    },
  },
  {
    request: {
      query: CreateUserDocument,
      variables: {
        input: {
          email: 'test@test.com',
          password: 'password',
        },
      } as CreateUserMutationVariables,
    },
    result: {
      data: {
        createUser: createUserResult,
      } as CreateUserMutation,
    },
  },
];

const renderComponent = (props?: LoginFormProps): RenderResult => {
  let cmp: RenderResult;

  act(() => {
    cmp = render(
      <MockedProvider mocks={mocks}>
        <LoginForm {...props} />
      </MockedProvider>
    );
  });

  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return cmp!;
};

const fillFields = async (container: HTMLElement) => {
  const username = container.querySelector('#username');
  const password = container.querySelector('#password');

  await act(async () => {
    await fireEvent.type(username, 'test@test.com', {
      delay: 10,
    });

    await fireEvent.type(password, 'password', {
      delay: 10,
    });
  });
};

describe('<LoginForm />', () => {
  it('should render without crashing', () => {
    const cmp = renderComponent();
    expect(cmp).toMatchSnapshot();
  });

  it('should handle login', async () => {
    const handleLogin = jest.fn();

    const { container } = renderComponent({
      afterLogin: handleLogin,
    });

    const loginBtn = container.querySelector('#login');

    await fillFields(container);

    await act(async () => {
      userEvent.click(loginBtn);
    });

    await waitFor(() => expect(handleLogin).toHaveBeenCalledTimes(1));
  });

  it('should handle signup', async () => {
    const handleCreate = jest.fn();

    const { container } = renderComponent({
      afterCreate: handleCreate,
      type: LoginFormType.Signup,
    });

    await fillFields(container);

    const signupBtn = container.querySelector('#signup');

    act(() => {
      userEvent.click(signupBtn);
    });

    await waitFor(() => expect(handleCreate).toHaveBeenCalledTimes(1));
    expect(handleCreate).toHaveBeenCalledWith(createUserResult);
    expect(useTokensStore.getState().tokens).toEqual(createUserResult.tokens);
  });
});
