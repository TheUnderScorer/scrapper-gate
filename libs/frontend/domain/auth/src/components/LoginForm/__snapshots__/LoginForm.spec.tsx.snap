// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="makeStyles-form-1"
      >
        <div
          class="css-kttiti-MuiStack-root"
        >
          <div
            class="makeStyles-alert-2 MuiBox-root css-0"
            style="opacity: 0; visibility: hidden;"
          />
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-l18d24-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="username"
              id="username-label"
            >
              E-mail
            </label>
            <div
              class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1vpizsw-MuiInputBase-root-MuiFilledInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                id="username"
                name="username"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-l18d24-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Password
            </label>
            <div
              class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1vpizsw-MuiInputBase-root-MuiFilledInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="makeStyles-action-3 css-guojqh-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary makeStyles-fab-4 css-2yq0w0-MuiButtonBase-root-MuiFab-root"
              id="login"
              tabindex="0"
              type="submit"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="makeStyles-form-1"
    >
      <div
        class="css-kttiti-MuiStack-root"
      >
        <div
          class="makeStyles-alert-2 MuiBox-root css-0"
          style="opacity: 0; visibility: hidden;"
        />
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-l18d24-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="username"
            id="username-label"
          >
            E-mail
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1vpizsw-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id="username"
              name="username"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-l18d24-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            Password
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1vpizsw-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="makeStyles-action-3 css-guojqh-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary makeStyles-fab-4 css-2yq0w0-MuiButtonBase-root-MuiFab-root"
            id="login"
            tabindex="0"
            type="submit"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
